modules/ROOT/pages/tuning-monitoring.adoc

```diff
@@ -0,0 +1,89 @@
+
+
+== CPU Resource
+Increased load can exhaust CPU resources. There are several ways to monitor these resources, two of them are:
+
+* Top command: `top` +
+The `%CPU` metric equals to the CPU usage divided by the number of cores.
+
+.Top command CPU stats
+image::mruntime-tuning-monitoring-top.png[Top command CPU stats]
+
+* Sar command: `sar -u  ALL $INTERVAL $COUNT` +
+Where `$INTERVAL` specifies the time interval to obtain the metrics, and `$COUNT` specifies the number of data points to collect. +
+Main metrics to pay attention are:
+
+** `%usr` shows how much CPU is used at the user level (application).
+** `%sys` shows how much CPU is used at system level (kernel).
+** `%idle` shows the percentage of time that the CPU was idle.
+
+.Sar command CPU stats
+image::mruntime-tuning-sar-cpu.png[Sar command CPU stats]
+
+== Memory Resource
+To understand how Mule uses memory, including heap and native memory or off-heap buffers, monitor the memory via several ways, two of them are:
+
+* GC Log +
+Enable GC (Garbage Collection) logging to obtain allocation rate, object growth, and so on. Additionally, obtain information about the activities in the heap. Notice that enabling GC logging has a light overhead on the disk.
+
+.CG Log wrapper.conf
+image::mruntime-tuning-wrapper.png[wrapper.conf]
+
+* Sar command: `sar -r $INTERVAL $COUNT` +
+Main metrics to pay attention are:
+
+** `kbmemfree` shows the amount of free memory available in kilobytes.
+** `kbmemused` shows the amount of used memory in kilobytes.
+** `kbcommit` shows the amount of memory in kilobytes needed for the current workload, which estimates how much RAM or swap is required to guarantee that it doesn't run out of memory.
+
+.Sar command memory stats
+image::mruntime-tuning-sar-memory.png[Sar command Memory stats]
+
+== Disk Resource
+Some Mule components log writing operations and scenarios with large payloads treated as repeatable streams exhibit, where the exhausted resource are I/O (input or output) requests. Monitor the disk resource using the Sar command `sar -d -p $INTERVAL $COUNT`.
+
+Main metrics to pay attention are:
+
+** `await` shows the average time in milliseconds for I/O requests issued to serve to the device to, which includes the time spent by the requests in the queue and the time spent servicing them.
+** `%util` helps to understand the saturation of the device showing the percentage of CPU time during which I/O requests are issued to bandwidth utilization for the device.
+
+.Sar command disk stats
+image::mruntime-tuning-sar-disk.png[Sar command Disk stats]
+
+== Network Resource
+One of the main purposes of Mule is to interconnect with several services that communicate by the net. Because of this, the network infrastructure can become a bottleneck and saturated. You need to know the bandwidth and monitor the utilization, a simple way to do it is using the Sar command `sar -n DEV $INTERVAL $COUNT`.
```

Comment:
```suggestion
Because Mule connects several services communicating through the internet, network use can become saturated and stall. To manage this, you need to know the available bandwidth and monitor the utilization. A simple way to do this is by using the Sar command `sar -n DEV $INTERVAL $COUNT`.
```